src/extension/csv-with-metadata-header/src/main/java/au/org/emii/geoserver/wfs/response/CSVWithMetadataHeaderOutputFormat.java

```diff
@@ -26,51 +14,168 @@
 import org.geoserver.platform.ServiceException;
 import org.geoserver.wfs.WFSGetFeatureOutputFormat;
 import org.geoserver.wfs.request.FeatureCollectionResponse;
-import org.geoserver.wfs.response.CSVOutputFormat;
+import org.geoserver.wfs.request.GetFeatureRequest;
+import org.geoserver.wfs.request.Query;
 import org.geotools.data.Transaction;
 import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.feature.type.DateUtil;
 import org.geotools.jdbc.JDBCDataStore;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.DateTimeFormatterBuilder;
+import org.joda.time.format.DateTimeParser;
+import org.joda.time.format.ISODateTimeFormat;
+import org.opengis.feature.Feature;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+
+import javax.xml.namespace.QName;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.text.NumberFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
 
 public class CSVWithMetadataHeaderOutputFormat extends WFSGetFeatureOutputFormat {
 
+    static final Pattern CSV_ESCAPES = Pattern.compile("[\"\n,\r]");
+
+    static final DateTimeParser[] parsers = {
+            DateTimeFormat.forPattern( "yyyy-MM-dd HH:mm:ss" ).getParser(),
+            DateTimeFormat.forPattern( "yyyy-MM-dd HH:mm:ss.S" ).getParser(),
+            DateTimeFormat.forPattern( "yyyy-MM-dd HH:mm:ss.SS" ).getParser(),
+            DateTimeFormat.forPattern( "yyyy-MM-dd HH:mm:ss.SSS" ).getParser()
+    };
+    static final DateTimeFormatter IMOS_DATE_TIME_FORMATTER = new DateTimeFormatterBuilder().append( null, parsers ).toFormatter();
+
     static Logger LOGGER = org.geotools.util.logging.Logging.getLogger("au.org.emii.geoserver.wfs.response");
 
     private Catalog catalog;
-    private CSVOutputFormat csvOutputFormat;
 
-    public CSVWithMetadataHeaderOutputFormat(GeoServer gs, Catalog catalog, CSVOutputFormat csvOutputFormat) {
+    public CSVWithMetadataHeaderOutputFormat(GeoServer gs, Catalog catalog) {
         super(gs, "csv-with-metadata-header");
         this.catalog = catalog;
-        this.csvOutputFormat = csvOutputFormat;
+        DateTimeZone.setDefault(DateTimeZone.UTC);
```

Comment:
This sets the default time zone for the JVM.  Best not to mess with this here - we are already specifying this as a system property when starting tomcat - specify the 'display' timezone when creating the datetime below - see comments below.